src/Nest/CommonAbstractions/Infer/Field/Field.cs

```diff
@@ -56,6 +56,7 @@ public Field(PropertyInfo property, double? boost = null, string format = null)
 		/// </remarks>
 		public string Format { get; set; }
 
+		// TODO: Rename to CacheableExpression in 8.0.0
```

Comment:
Why not introduce it now, put an obsolete attribute on this and have the getter and setter populate the new property?